fix: The Fee details are not fetched in Program Enrollment

mujeerhashmi opened the pull request:
The Fee details are not fetched properly into program enrollment based on
program, academic year, and student category. It has been fetched &
populated properly now.

Signed-off-by: Syed Mujeer Hashmi <mujeerhashmi@4csolutions.in>

<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/erpnext/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

ruchamahabal commented on erpnext/education/api.py:
```diff
@@ -198,22 +198,6 @@ def get_fee_components(fee_structure):
 
 
-@frappe.whitelist()
-def get_fee_schedule(program, student_category=None):
```
Let's not remove this API since it might be used in implementations. It's also used in demo creation:

https://github.com/frappe/erpnext/blob/7767dc1ee3536c72ae5db7eafaa387e07a7e9ac2/erpnext/demo/user/education.py#L51

ruchamahabal commented on erpnext/education/doctype/program_enrollment/program_enrollment.js:
```diff
@@ -57,15 +57,22 @@ frappe.ui.form.on('Program Enrollment', {
-				method: 'erpnext.education.api.get_fee_schedule',
+				method: 'frappe.client.get_list',
 				args: {
-					'program': frm.doc.program,
-					'student_category': frm.doc.student_category
+					doctype: 'Fee Schedule',
+					filters: {
+						'program': frm.doc.program,
+						'student_category': frm.doc.student_category,
+						'academic_year': frm.doc.academic_year
+					},
+					fields: ['academic_term', 'fee_structure', 'student_category', 'due_date', 'total_amount as amount']
 				},
 				callback: function(r) {
 					if (r.message) {
+						cur_frm.clear_table("fees");
+						frm.refresh_fields('fees');
 						frm.set_value('fees' ,r.message);
-						frm.events.get_courses(frm);
+						frm.refresh_fields('fees');
 					}
 				}
 			});
```
Can you move this implementation to the server-side to make it API friendly?

https://github.com/frappe/erpnext/wiki/Coding-Standards#all-business-logic-must-be-api-friendly-and-must-be-implemented-in-server-side-python

You can refer [`get_program_courses`](https://github.com/mujeerhashmi/erpnext/blob/4defc7593cf96b602bbd5fb36d16131e632da94e/erpnext/education/doctype/program_enrollment/program_enrollment.py#L188-L204) or `get_students` in `program_enrollment.py` for the implementation.

mujeerhashmi replied:
This is specific to program enrollment. So I think this is okay,

mujeerhashmi committed:
fix: The Fee details are not fetched in Program Enrollment

The Fee details are not fetched properly into program enrollment based on
program, academic year, and student category. It has been fetched &
populated properly now.

Signed-off-by: Syed Mujeer Hashmi <mujeerhashmi@4csolutions.in>

mujeerhashmi commented:
@ruchamahabal kindly review

mujeerhashmi commented:
Keep alive

mujeerhashmi commented:
@ruchamahabal kindly review.

mujeerhashmi commented:
@ruchamahabal kindly review

mujeerhashmi commented:
@ruchamahabal can we get this merged please so that it will make it to the next release. I have already fixed your review comments and hope it should not take more time.
